spinalcordtoolbox/deepseg/models.py

```diff
@@ -350,16 +351,40 @@ def is_valid(path_models):
     :param path_models: str or list: Absolute path(s) to folder(s) that enclose the model files.
     """
     def _is_valid(path_model):
-        name_model = path_model.rstrip(os.sep).split(os.sep)[-1]
-        return (os.path.exists(os.path.join(path_model, name_model + '.pt')) or
-                os.path.exists(os.path.join(path_model, name_model + '.onnx'))) and os.path.exists(
-            os.path.join(path_model, name_model + '.json'))
+        return has_ivadomed_files(path_model) or has_ckpt_files(path_model) or has_pth_files(path_model)
```

Comment:
Good question!! 

I originally implemented model ensembling on request for single ivadomed model -- `model_seg_lesion_mp2rage` (see https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4345#discussion_r1457981736). The team had already trained the various seeds, and just needed the support for averaging the outputs.

Since then, AFAIK no new models have been trained with ensembling in mind. Meaning, if we _were_ to create new models moving forward, I would presume they would all be done under the umbrella of a single framework (i.e. not mixing ivadomed/monai).

Reply:
That said, I haven't extended ensembling support to monai/nnunet yet?

Doing so would likely require another small rewrite (i.e. splitting up `segment_and_average_volumes`) such that the averaging process could be performed on any of the frameworks.

Reply:
Ah ha, I think you gave me all the relevant info to answer the question:
* we don't currently need much ensembling support (because it's just one model), and
* it _would_ require some rewriting work to support mixed-backend ensembling.

Given that, I would suggest sticking to single-backend ensembling for now in `_is_valid(path_model)`.